edx_sga/tests/test_sga.py

```diff
@@ -0,0 +1,617 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for SGA
+"""
+import datetime
+import json
+import mimetypes
+import unittest
+import tempfile
+import mock
+import pkg_resources
+import pytz
+
+from ddt import ddt, data  # pylint: disable=import-error
+from django.conf import settings  # lint-amnesty, pylint: disable=import-error
+from django.core.exceptions import PermissionDenied
+from django.core.files.storage import FileSystemStorage
+from opaque_keys.edx.locations import Location  # lint-amnesty, pylint: disable=import-error
+from opaque_keys.edx.locator import CourseLocator  # lint-amnesty, pylint: disable=import-error
+from xblock.field_data import DictFieldData
+from xblock.fields import DateTime
+
+from edx_sga.tests.common import DummyResource, DummyUpload
+
+
+SHA1 = 'da39a3ee5e6b4b0d3255bfef95601890afd80709'
+
+
+def fake_get_submission(upload):
+    """retrurns fake submission"""
+    return {
+        "answer": {
+            "sha1": SHA1,
+            "filename": upload.file.name.encode('utf-8'),
+            "mimetype": mimetypes.guess_type(upload.file.name.encode('utf-8'))[0]
+        }
+    }
+
+
+class MockedStudentModule(object):
+    """dummy representation of xblock class"""
+    def __init__(self):
+        self.course_id = CourseLocator(org='foo', course='baz', run='bar')
+        self.module_state_key = "foo"
+        self.student = mock.Mock(username="fred6", is_staff=False, password="test")
+        self.state = '{"display_name": "Staff Graded Assignment"}'
+
+    def save(self):
+        """save method do nothing"""
+        pass
+
+
+@ddt
+class StaffGradedAssignmentMockedTests(unittest.TestCase):
+    """
+    Create a SGA block with mock data.
+    """
+    def setUp(self):
+        """
+        Creates a test course ID, mocks the runtime, and creates a fake storage
+        engine for use in all tests
+        """
+        super(StaffGradedAssignmentMockedTests, self).setUp()
+        self.course_id = CourseLocator(org='foo', course='baz', run='bar')
+        self.runtime = mock.Mock(anonymous_student_id='MOCK')
+        self.scope_ids = mock.Mock()
+        tmp = tempfile.mkdtemp()
+        patcher = mock.patch(
+            "edx_sga.sga.default_storage",
+            FileSystemStorage(tmp))
+        patcher.start()
+        self.addCleanup(patcher.stop)
+        self.staff = mock.Mock(return_value={
+            "password": "test",
+            "username": "tester",
+            "is_staff": True
+        })
+
+    def make_one(self, display_name=None, **kw):
```

Comment:
Please change to `kwargs` to match the name used for other functions. Please also rename `make_one` to `make_xblock`